src/core/Akka/Actor/Scheduler/HashedWheelTimerScheduler.cs

```diff
@@ -215,31 +235,58 @@ private long WaitForNextTick()
             {
                 for (;;)
                 {
-                    long currentTime = HighResMonotonicClock.Ticks - _startTime;
-                    var sleepMs = ((deadline - currentTime + TimeSpan.TicksPerMillisecond - 1) / TimeSpan.TicksPerMillisecond);
-
-                    if (sleepMs <= 0) // no need to sleep
+                    var currentTime = HighResMonotonicClock.Ticks - _startTime;
+                    var ticksToSleep = deadline - currentTime;
+                    
+                    if (ticksToSleep <= 0) // no need to sleep
                     {
                         if (currentTime == long.MinValue) // wrap-around
                             return -long.MaxValue;
+                        
                         return currentTime;
-
                     }
 
+                    var stopWatch = Stopwatch.StartNew();
```

Comment:
Instead of starting a new `Stopwatch` here, if you use `HighRestMonotonicClock.Ticks` that will give you the current `Stopwatch.Elapsed` value since Akka.NET started and you can acquire a second comparison value where you have `Stopwatch.Sleep` instead - just to save on allocations.

Reply:
Ah, actually had to move this `stopwatch` to private field instantiated once - but missed this. And sure, your suggestion is even better.

Reply:
Used the same thing for sleep time tracking when making `Thread.SpinWait`, instead of separate `Stopwatch` instance usage.